packages/admin-ui/src/app/transfers/transfers.component.ts

```diff
@@ -1,31 +1,68 @@
-import {Component, OnDestroy, OnInit} from "@angular/core";
-import {TransfersService} from "src/app/_services_and_types/transfers.service";
-import {BehaviorSubject, Subscription} from "rxjs";
-import {Transfer} from "src/app/_services_and_types/transfer_types";
-import {MessageService} from "src/app/_services_and_types/message.service";
-import {UnauthorizedError} from "src/app/_services_and_types/errors";
-import {paginate, PaginateResult} from "../_utils";
+import { Component, OnDestroy, OnInit } from "@angular/core";
+import { TransfersService } from "src/app/_services_and_types/transfers.service";
+import { ParticipantsService } from "src/app/_services_and_types/participants.service";
+import { BehaviorSubject, Subscription } from "rxjs";
+import { TransfersSearchResults } from "src/app/_services_and_types/transfer_types";
+import { MessageService } from "src/app/_services_and_types/message.service";
+import { UnauthorizedError } from "src/app/_services_and_types/errors";
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { HUB_PARTICIPANT_ID, IParticipant } from "@mojaloop/participant-bc-public-types-lib";
 
 @Component({
 	selector: 'app-transfers',
-	templateUrl: './transfers.component.html'
+	templateUrl: './transfers.component.html',
+	styleUrls: ['./transfers.component.css']
 })
 export class TransfersComponent implements OnInit, OnDestroy {
 
 	readonly ALL_STR_ID = "(All)";
-	transfers: BehaviorSubject<Transfer[]> = new BehaviorSubject<Transfer[]>([]);
+	public criteriaFromDate = "";
+
+	transfers: BehaviorSubject<TransfersSearchResults> = new BehaviorSubject<TransfersSearchResults>({
+		pageSize: 10,
+		totalPages: 1,
+		pageIndex: 0,
+		items: []
+	});
 	transfersSubs?: Subscription;
+	participantsSubs?: Subscription;
+	filterForm: FormGroup;
+	pageIndex: number = 0;
+	pageSize: number = 10;
+	userPageIndex: number = 1; //for UI control to show current page number
 
+	//Filters
 	keywordState: BehaviorSubject<string[]> = new BehaviorSubject<string[]>([]);
 	keywordCurrency: BehaviorSubject<string[]> = new BehaviorSubject<string[]>([]);
 	keywordSourceAppName: BehaviorSubject<string[]> = new BehaviorSubject<string[]>([]);
 	keywordsSubs?: Subscription;
 
-	paginateResult: BehaviorSubject<PaginateResult | null> = new BehaviorSubject<PaginateResult | null>(null);
-
-	public criteriaFromDate = "";
+	//Filters
+	participants: BehaviorSubject<IParticipant[]> = new BehaviorSubject<IParticipant[]>([]);
+	partyNameList = [this.ALL_STR_ID, "VisionFund_Myanmar", "OkDollar", "Company_A_Limited"];
```

Comment:
Is there a way to have this in a non-hardcoded way at the moment? If not, please add a TODO otherwise it becomes way too specific to a business model.